src/raft.c

```diff
@@ -366,13 +386,17 @@ void RaftExecuteCommandArray(RedisRaftCtx *rr,
 
     /* When we're in cluster mode, go through handleSharding. This will perform
      * hash slot validation and return an error / redirection if necessary. */
-    if (handleSharding(rr, reply_ctx, cmds) != RR_OK) {
-        return;
+    if (handleSharding(rr, req ? req->ctx : NULL, cmds) != RR_OK) {
+        return NULL; /* sharding error, so even if blocking command, don't */
     }
 
-    ClientSession *client_session = getClientSession(rr, cmds, reply_ctx != NULL);
+    ClientSession *client_session = getClientSession(rr, cmds, req != NULL);
     (void) client_session; /* unused for now */
 
+    if (cmds->cmd_flags & CMD_SPEC_BLOCKING) {
+        RedisModule_Assert(RaftRedisReplaceBlockingTimeout(cmds) == REDISMODULE_OK);
```

Comment:
As far as I can see, you don't use return value of this function. So, maybe we can assert things inside of it. 
I

Reply:
ok, adjusted